### ISSUE TITLE
Implement a matvec method for the DEN matrix class

### ISSUE THREAD

**Issue by KyleVaughn** [bkochuna/ners570f24-SpMV#12]
## Description:
Create a matrix-vector multiplication method for the DEN matrix class. 

## Definition of done:

**Comment by athenafhl** [comment]
I'm reviewing this.

**Comment by dbloc-umich** [comment]
@athenafhl  this is ready for review.

**Comment by athenafhl** [comment]
Looks good for me

**Comment by athenafhl** [comment]
@dbloc-umich Hi -- Did you work in the DEN branch and use Den PR? I'm a little bit confused since I didn't see your name in the commits.

**Comment by dbloc-umich** [comment]
@athenafhl yeah, but apparently the commit was under my other account @lduong1 for some reason.
